lib/src/domain/usecase/implementation/get_movies_categories_usecase.dart

```diff
@@ -0,0 +1,23 @@
+import '../../../core/parameter/movie_list_parameters.dart';
+import '../../../core/util/data_state.dart';
+import '../../../core/util/enums.dart';
+import '../../../data/repository/movie_repository.dart';
+import '../../entity/movie.dart';
+import '../usecase_interface.dart';
+
+class GetMoviesCategoriesUsecase
+    implements UseCase<DataState<List<Movie>>, MovieListParameters> {
+  late final MovieRepository repository;
+
+  GetMoviesCategoriesUsecase({required this.repository});
+
+  @override
+  Future<DataState<List<Movie>>> call({
+    MovieListParameters? params,
+    required Endpoint endpoint,
```

Comment:
If you know that always you are get the movies by categories, you need to specify the endpoint?